jax/experimental/checkify.py

```diff
@@ -495,16 +564,65 @@ def assert_discharge_rule(error, enabled_errors, pred, code, *, msgs):
 
 ## checkify api
 
-ErrorCategory = enum.Enum('ErrorCategory', ['NAN', 'OOB', 'DIV', 'ASSERT'])
+ErrorCategory = enum.Enum('ErrorCategory', ['NAN', 'OOB', 'DIV', 'USER_CHECK'])
 
-float_errors = {ErrorCategory.NAN, ErrorCategory.DIV}
-index_errors = {ErrorCategory.OOB}
+float_errors = frozenset({ErrorCategory.NAN, ErrorCategory.DIV})
+index_errors = frozenset({ErrorCategory.OOB})
 automatic_errors = float_errors | index_errors
-user_asserts = {ErrorCategory.ASSERT}
+user_checks = frozenset({ErrorCategory.USER_CHECK})
 
 Out = TypeVar('Out')
-def checkify(fun: Callable[..., Out], errors: Set[ErrorCategory] = user_asserts
+
+
+def checkify(fun: Callable[..., Out],
+             errors: FrozenSet[ErrorCategory] = user_checks
              ) -> Callable[..., Tuple[Error, Out]]:
+  """Check for run-time errors in ``fun``.
+
+  Run-time errors are either user-added ``checkify.check`` assertions, or
+  automatically added checks like NaN checks, depending on the ``errors``
+  argument.
+
+  The returned function will return an Error object along with the output of
+  the function. ``err.get()`` will either return ``None`` (if no error occurred)
+  or a string containing an error message. This error message will correspond to
+  the first error which occurred.
+
+  The kinds of errors are:
+    - ErrorCategory.USER_CHECK: a ``checkify.check`` predicate evaluated
```

Comment:
Why make these all-caps, like `USER_CHECK` instead of `user_check`?

Reply:
They are the enum values, which I think style-wise [are dictated to be upper case](https://mail.python.org/pipermail/python-ideas/2016-September/042340.html)? 

It probably makes more sense to refer to the sets here instead of the enum values, because those are the ones which need to be passed into the function. Maybe every enum value should have a corresponding (camel_case) set?

Reply:
Ah thanks for teaching me about that style guidance. (I always feel like all-caps variables are yelling at me...)

Having a corresponding singleton set for each case makes sense... so does perhaps making this function flexible enough to handle both (i.e. a raw enum value as well as sets). We can fiddle with that after adding!